src/zino/snmp.py

```diff
@@ -499,11 +500,26 @@ def udp_transport_target(self) -> Union[UdpTransportTarget, Udp6TransportTarget]
 
 def _convert_varbind(ident: ObjectIdentity, value: ObjectType) -> SNMPVarBind:
     """Converts a PySNMP varbind pair to an Identifier/value pair"""
-    mib, obj, row_index = ident.getMibSymbol()
+    mib, obj, indices = ident.getMibSymbol()
     value = _mib_value_to_python(value)
+    row_index = reduce(lambda acc, index: acc + _index_to_tuple(index), indices, ())
     return Identifier(mib, obj, OID(row_index)), value
 
 
+def _index_to_tuple(index: ObjectIdentity) -> Tuple[int, ...]:
+    """Unrolls a PySNMP index object to a tuple of integers"""
+    try:
+        return tuple(index)
+    except TypeError:
+        try:
+            return index.asTuple()
+        except AttributeError:
+            try:
+                return index.getOid()
+            except AttributeError:
+                return (int(index),)
+
+
```

Comment:
Upon testing the new SNMP framework I've found that this still isn't up to snuff, as some bits keep getting cut off.  Notably, an object like ` .1.3.6.1.4.1.9.9.187.1.2.5.1.4.1.4.10.0.0.1` would normally be broken down to `CISCO-BGP4-MIB::cbgpPeer2AdminStatus.ipv4."10.0.0.1"`, but what we want from our function is something more like `CISCO-BGP4-MIB::cbgpPeer2AdminStatus.1.4.10.0.0.1`.   I.e. the index returned should be `1.4.10.0.0.1` (`1` is the type of address represented, `4` is the number of subsequent octets that represent the address) - instead this only return `4.10.0.0.1`.  Only the `bgpstatemonitortask` cares right now, but this PR makes it expect the latter.

Reply:
I've pushed a new commit where the strategy becomes more or less the same as in the netsnmp-cffi replacement branch: Reconstruct a prefix from the mib and object name, and then strip that from the incoming OID argument to get the row_index suffix value.